pycqed/analysis_v2/syndrome_analysis.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import numpy as np
+import pycqed.analysis_v2.base_analysis as ba
+import pycqed.analysis.tools.data_manipulation as dm_tools
+
+
+class Single_Qubit_RoundsToEvent_Analysis(ba.BaseDataAnalysis):
+
+    def __init__(self, t_start: str=None, t_stop: str=None,
+                 data_file_path: str=None,
+                 options_dict: dict=None, extract_only: bool=False,
+                 do_fitting: bool=True, auto=True):
+        super().__init__(t_start=t_start, t_stop=t_stop,
+                         data_file_path=data_file_path,
+                         options_dict=options_dict,
+                         extract_only=extract_only, do_fitting=do_fitting)
+        self.single_timestamp = False
+        exp_meta_str = 'Experimental Data.Experimental Metadata.'
+        self.params_dict = {
+            'measurementstring': 'measurementstring',
+            'measured_values': 'measured_values',
+            'depletion_time': exp_meta_str+'depletion_time',
+            'net_gate': exp_meta_str+'net_gate',
+            'sequence_type': exp_meta_str+'sequence_type',
+            'feedback': exp_meta_str+'feedback'}
+
+        self.numeric_params = []
+        if auto:
+            self.run_analysis()
+
+    def process_data(self):
+        """
+        """
+        # N.B. flipping is the default (net_gate = pi-pulse)
+        net_pulse_pat = 'flipping'
+        if not len(set(self.raw_data_dict['net_gate'])) == 1:
+            logging.warning('Different net pulses in dataset.')
+        if self.raw_data_dict['net_gate'][0] == 'i':
+            net_pulse_pat = 'constant'
+        if self.raw_data_dict['feedback'][0]:
+            net_pulse_pat = 'FB_to_ground'
+
+        exp_pattern = self.options_dict.get('exp_pattern', net_pulse_pat)
+
+        raw_dat = self.raw_data_dict['measured_values']
+        nr_expts = np.shape(raw_dat)[0]
+        raw_pat = [raw_dat[i][0] for i in range(nr_expts)]
+
+        s_events = []
+        d_events = []
+        for i in range(nr_expts):
+            err_marker = getattr(dm_tools,
+                                 'mark_errors_{}'.format(exp_pattern))
```

Comment:
Elegant but can lead to the most confusing errors...
I  would put an explicit 

```
if exp_pattern == 'i':
    dm_tools.mark_errors_i(...)
elif exp_pattern == 'pi':
    dm_tools.mark_errors_pi(...)
```

etc